src/skeleplex/graph/skeleton_graph.py

```diff
@@ -50,6 +50,52 @@ def skeleton_graph_decoder(json_object):
     return json_object
 
 
+def make_graph_directed(graph: nx.Graph, origin=int) -> nx.DiGraph:
+    """Return a directed graph from an undirected graph.
+
+    The directed graph has the same nodes and edges as the undirected graph.
+
+    Parameters
+    ----------
+    graph : nx.Graph
+        The undirected graph to convert to a directed graph.
+    origin : int
+        The node to use as the origin node for the directed graph.
+        The origin node will have no incoming edges.
+    """
+    if isinstance(graph, nx.DiGraph):
+        print("The input graph is already a directed graph.")
+        return graph
+    if len(list(nx.connected_components(graph))) > 1:
+        Warning("""
+        The input graph is not connected.
+        The unconnected components might lose edges
+        """)
+        origin_part = nx.node_connected_component(graph, origin)
+        fragments = graph.subgraph(set(graph.nodes()) - origin_part)
+        graph = graph.subgraph(origin_part)
+    else:
+        fragments = None
+
+    di_graph = nx.DiGraph(graph)
+    di_graph.remove_edges_from(di_graph.edges - nx.bfs_edges(di_graph, origin))
+
+    if fragments:
+        # choose a node with the highest degree as the origin node
+        # This could lead to the wrong edges being removed, but it is a good start
+        for fragment in nx.connected_components(fragments):
+            fragment_subgraph = fragments.subgraph(fragment)
+            origin = max(fragment_subgraph.degree, key=lambda x: x[1])[0]
```

Comment:
can you please add a comment explaining this line?

Reply:
Done, this approach is for sure not optimal, but I don't see an easy solution. Alternatively, we could store two edges for each branch in the fragments, but I imaging that this gets weird when we draw the tree in the viewer.

Reply:
Yeah, makes sense. Thanks